src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs

```diff
@@ -95,51 +100,64 @@ private byte[] GetPartitioningKey(IEnumerable<MessageHeader> headers)
 
         private Confluent.Kafka.IProducer<byte[], byte[]> CreateInnerProducer()
         {
-            _logger.LogDebug("Creating Confluent.Kafka.Producer...");
-
-            return new Confluent.Kafka.ProducerBuilder<byte[], byte[]>(Endpoint.Configuration.ConfluentConfig).Build();
+            _logger.LogDebug("Creating Confluent.Kafka.Producer...");
+
+            return new Confluent.Kafka.ProducerBuilder<byte[], byte[]>(Endpoint.Configuration.ConfluentConfig)
+                .SetStatisticsHandler((_, statistics) =>

```

Comment:
The PR title reads "publish statistics *and errors* from KafkaProducer". Did you forget to bind the error handler (or got lost in the rebase)?

Reply:
I deleted it due to our conversation: I have no use case for it and you mentioned that all errors are handled by silverback anyway. Did we talk past each other?